From ad80642e5ff3dbbb9e3f2f353e4684eda7086261 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 13 Jul 2021 13:19:36 +0300
Subject: [PATCH 04/49] can: llce: move llce can utils into a new header

commit ad80642e5ff3dbbb9e3f2f353e4684eda7086261 from
https://source.codeaurora.org/external/autobsps32/linux

unpack_word0 and unpack_word1 functions will be used by
llce logger.

Upstream-Status: Pending

Issue: ALB-5830, ALB-7344
Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/can/llce_can.c                    | 26 +-------------
 .../linux/mailbox/nxp-llce/llce_can_utils.h   | 35 +++++++++++++++++++
 2 files changed, 36 insertions(+), 25 deletions(-)
 create mode 100644 include/linux/mailbox/nxp-llce/llce_can_utils.h

diff --git a/drivers/net/can/llce_can.c b/drivers/net/can/llce_can.c
index b6d211424af1..baf1a3a2fd2b 100644
--- a/drivers/net/can/llce_can.c
+++ b/drivers/net/can/llce_can.c
@@ -5,6 +5,7 @@
 #include <linux/ctype.h>
 #include <linux/kernel.h>
 #include <linux/mailbox/nxp-llce/llce_can.h>
+#include <linux/mailbox/nxp-llce/llce_can_utils.h>
 #include <linux/mailbox/nxp-llce/llce_mailbox.h>
 #include <linux/mailbox_client.h>
 #include <linux/mailbox_controller.h>
@@ -505,31 +506,6 @@ static void llce_tx_notif_callback(struct mbox_client *cl, void *msg)
 	netif_wake_queue(llce->can.dev);
 }
 
-static void unpack_word0(u32 word0, bool *rtr, bool *ide,
-			 u32 *std_id, u32 *ext_id)
-{
-	if (word0 & LLCE_CAN_MB_IDE) {
-		*ide = true;
-		*ext_id = (word0 & CAN_EFF_MASK) >> CAN_SFF_ID_BITS;
-		*std_id = word0 & CAN_SFF_MASK;
-	} else {
-		*ide = false;
-		*std_id = (word0 & LLCE_CAN_MB_IDSTD_MASK) >>
-			LLCE_CAN_MB_IDSTD_SHIFT;
-	}
-
-	*rtr = !!(word0 & LLCE_CAN_MB_RTR);
-}
-
-static void unpack_word1(u32 word1, bool *fdf, u8 *len, bool *brs,
-			 bool *esi)
-{
-	*len = word1 & LLCE_CAN_MB_DLC_MASK;
-	*brs = !!(word1 & LLCE_CAN_MB_BRS);
-	*esi = !!(word1 & LLCE_CAN_MB_ESI);
-	*fdf = !!(word1 & LLCE_CAN_MB_FDF);
-}
-
 static int send_rx_msg(struct llce_can *llce, struct llce_rx_msg *msg)
 {
 	int ret = mbox_send_message(llce->rx, msg);
diff --git a/include/linux/mailbox/nxp-llce/llce_can_utils.h b/include/linux/mailbox/nxp-llce/llce_can_utils.h
new file mode 100644
index 000000000000..15a949fc6ca4
--- /dev/null
+++ b/include/linux/mailbox/nxp-llce/llce_can_utils.h
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright 2021 NXP
+ */
+#ifndef LLCE_CAN_UTILS_H
+#define LLCE_CAN_UTILS_H
+
+#include <linux/ctype.h>
+
+static inline void unpack_word0(u32 word0, bool *rtr, bool *ide,
+				u32 *std_id, u32 *ext_id)
+{
+	if (word0 & LLCE_CAN_MB_IDE) {
+		*ide = true;
+		*ext_id = (word0 & CAN_EFF_MASK) >> CAN_SFF_ID_BITS;
+		*std_id = word0 & CAN_SFF_MASK;
+	} else {
+		*ide = false;
+		*std_id = (word0 & LLCE_CAN_MB_IDSTD_MASK) >>
+		    LLCE_CAN_MB_IDSTD_SHIFT;
+	}
+
+	*rtr = !!(word0 & LLCE_CAN_MB_RTR);
+}
+
+static inline void unpack_word1(u32 word1, bool *fdf, u8 *len, bool *brs,
+				bool *esi)
+{
+	*len = word1 & LLCE_CAN_MB_DLC_MASK;
+	*brs = !!(word1 & LLCE_CAN_MB_BRS);
+	*esi = !!(word1 & LLCE_CAN_MB_ESI);
+	*fdf = !!(word1 & LLCE_CAN_MB_FDF);
+}
+
+#endif
-- 
2.17.1

